### PR TITLE
[DOC-12689] Adding details to Sitecore Content Hub

### PR DESCRIPTION
**Pull request by dberubecoveo**
On behalf of Michel Deslandes

### CODE REVIEW

**Review by marilynechan — APPROVED** [review]

File: Sitecore Content Hub/README.md [hunk 1]
```diff
@@ -15,11 +15,82 @@ To fully understand how to use this example, you must:
 3. Under **Sources**, [create a Generic REST API source](https://docs.coveo.com/en/1896/):
```
Comment (mdeslandescoveo) [inline Sitecore Content Hub/README.md:15]:
```suggestion
3. Under **Sources**, [create a REST API source](https://docs.coveo.com/en/1896/):
```

File: Sitecore Content Hub/README.md [hunk 2]
```diff
@@ -15,11 +15,82 @@ To fully understand how to use this example, you must:
...
+
 ## Reference
-[Sitecore Content Hub's REST API documentation](https://doc.sitecore.com/ch/en/developers/42/cloud-dev/rest-api--about.html)
\ No newline at end of file
+
+[Sitecore Content Hub REST API documentation](https://doc.sitecore.com/ch/en/developers/42/cloud-dev/rest-api--about.html)
+
+[Generic REST API source concepts](https://docs.coveo.com/en/3131/)
+
+[Generic REST API source reference](https://docs.coveo.com/en/1525/)
```
Comment (mdeslandescoveo) [inline Sitecore Content Hub/README.md:96]:
```suggestion
[REST API source reference](https://docs.coveo.com/en/1525/)
```

**Review by mdeslandescoveo — APPROVED** [review]
Could we please also remove the word "Generic" in lines 1 and 10?

File: Sitecore Content Hub/README.md (hunk 1 above)
Comment (dberubecoveo) [reply Sitecore Content Hub/README.md]: Done

File: Sitecore Content Hub/README.md [hunk 3]
```diff
@@ -15,11 +15,82 @@ To fully understand how to use this example, you must:
...
 5. Check whether your source indexes the desired content properly. You might find you need an [indexing pipeline extension](https://docs.coveo.com/en/1645/) to achieve the expected result.
 
+## Understanding the `SourceJSONConfig.json` file
+
+The following are a few pointers to help you understand the example JSON configuration:
+
+### Authenticating calls to the Content Hub API
+
+All calls to the Sitecore Content Hub performed in the configuration [inherit](https://docs.coveo.com/en/3131/#inheritable-properties) the `X-Auth-Token` header property.
+The `@ApiKey` syntax is used to retrieve the `API key` field value from the Generic REST source user interface [Authentication section](https://docs.coveo.com/en/1896/#authentication-section).
```
[the remark this answers is not in the document]
Comment (dberubecoveo) [reply Sitecore Content Hub/README.md]: Done

File: Sitecore Content Hub/README.md [hunk 4]
```diff
@@ -15,11 +15,82 @@ To fully understand how to use this example, you must:
...
+
+Use your custom metadata to [populate fields](https://docs.coveo.com/en/1896/#completion).
+These fields can then be used in search interface [facets](https://docs.coveo.com/en/198/) and [result templates](https://docs.coveo.com/en/atomic/latest/reference/result-template-components/atomic-result-fields-list/) to enhance the search experience.
+
+---
+
+### Specifying whether to index the main GET call items
+
+No item from the `endpoints` level is actually indexed with [`ActionOnItem`](https://docs.coveo.com/en/1525/#actiononitem-string-required) set to `Ignore` and no [`Condition`](https://docs.coveo.com/en/1525/#condition-string) specified.
+Only [some sub-items are indexed](#indexing-sub-items-on-a-conditional-basis) in the Generic REST API source.
```
[the remark this answers is not in the document]
Comment (dberubecoveo) [reply Sitecore Content Hub/README.md]: Done

File: Sitecore Content Hub/README.md [hunk 5]
```diff
@@ -15,11 +15,82 @@ To fully understand how to use this example, you must:
...
+
+<img src="images\GenericRestSubItemsActionOnItem.png" width="600" alt="Sub-items ActionOnItem value | Coveo">
+
 ## Reference
-[Sitecore Content Hub's REST API documentation](https://doc.sitecore.com/ch/en/developers/42/cloud-dev/rest-api--about.html)
\ No newline at end of file
+
+[Sitecore Content Hub REST API documentation](https://doc.sitecore.com/ch/en/developers/42/cloud-dev/rest-api--about.html)
+
+[Generic REST API source concepts](https://docs.coveo.com/en/3131/)
```
[the remark this answers is not in the document]
Comment (dberubecoveo) [reply Sitecore Content Hub/README.md]: Done

File: Sitecore Content Hub/README.md (hunk 2 above)
Comment (dberubecoveo) [reply Sitecore Content Hub/README.md]: Done

**Pull request merged**